[nativert] clangtidy

dolpm opened the pull request:
Differential Revision: D80480826

Skylion007 commented:
> @cyyever any idea why these rules aren't being applied?

Rules are only applied to header files included by .cpp file. To apply this rules, you would need to have a cpp file include the header because a header is not valid until it's included and clang-tidy cannot do static analysis on it. We hit this issue with pybind11. We resolved it there by having clang-tidy apply to all the cpp tests and then use header-filter regex to apply it to the pybind11 headers.

dolpm committed:
[nativert] clangtidy (#160915)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/160915

Test Plan:
ci

Rollback Plan:

Differential Revision: D80480826

Skylion007 commented on torch/nativert/.clang-tidy:
```diff
@@ -0,0 +1,59 @@
+---
```
You don't want to inherit parent config?

Skylion007 commented on torch/nativert/.clang-tidy:
```diff
@@ -0,0 +1,59 @@
+---
+Checks: '-*,readability-identifier-naming,'
```
You want to disable all the useful checks? Like performance?
